Containerfile.tests

```diff
@@ -0,0 +1,26 @@
+#
+# Copyright (c) 2023 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+#
+
+FROM docker.io/golang:1.21 AS builder
```

Comment:
Can we use RHEL as the base image, like we do in [Containerfile](https://github.com/openshift-kni/oran-o2ims/blob/b72277e24d36979f830ee8a44ef49fe95cf0fd26/Containerfile#L15C1-L30C13)?

Reply:
We can, for sure. I wanted to get away from having to install the Go 1.21 compiler and I saw that other operators in openshift-kni are also using this image. The preferred one would be the Go 1.21 version of registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.20-openshift-4.15, but there isn't one yet.